### DISCUSSION TITLE
Choice hover sound effect with arrow keys

### DISCUSSION THREAD

**Discussion by nookstray** [dialogic-godot/dialogic#852] [Q&A] [answered]
I asked this before but i wanted to know how I could make the choice hover sound effect work with the arrow keys instead of the mouse. I know it’s something I would need to code, but I don’t know where to start...

**Comment by Jowan-Spooner** [comment, marked as answer]
Hi you will find all the relevant functions in the DialogNode, which is massive, so it's very understandable you didn't know where to begin. 

The function that calls the sound is the `_on_option_focused` function (around line 1220).

You will want to call this function when the button get's focused, so you will want to connect the buttons 'focus_entered' signal to that function when the choice button is created. 

You can do so in the `add_choice_button` function (around line 1170 or so).

It should look something like this I think: 
`button.connect('focus_entered', self, '_on_option_focused')`

Hope this helps.

**Reply by nookstray** [reply]
Hello! I found both of the functions that you mentioned, but I’m not exactly sure where to add the code. I need to add it here, correct?

```
func add_choice_button(option: Dictionary) -> Button:
	var button = get_classic_choice_button(option['label'])
	button_container.set('custom_constants/separation', current_theme.get_value('buttons', 'gap', 20))
	button_container.add_child(button)
```

If so, where exactly would this line of code go?

`button.connect('focus_entered', self, '_on_option_focused')`

**Reply by nookstray** [reply]
Nevermind, it works! Just had to add it at the end like this:

```
func add_choice_button(option: Dictionary) -> Button:
	var button = get_classic_choice_button(option['label'])
	button_container.set('custom_constants/separation', current_theme.get_value('buttons', 'gap', 20))
	button_container.add_child(button)
	button.connect('focus_entered', self, '_on_option_focused')
```

Thanks for everything!
